src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1292,4 +1293,1364 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+                                       capacityModifierFuncTemp);
+        capacityModifierFuncTemp = 0.0;
+    }
+
+    Real64 availableCapacity = this->referenceCapacity * capacityModifierFuncTemp;
+    Real64 partLoadRatio = 0.0;
+    if (availableCapacity > 0) {
+        partLoadRatio = max(0.0, min(std::abs(FFHPloadSideLoad) / availableCapacity, 1.0));
+    }
+
+    // evaluate the actual current operating load side heat transfer rate
+
+    // this->loadSideHeatTransfer = availableCapacity * partLoadRatio;
+    this->loadSideHeatTransfer = availableCapacity * (partLoadRatio >= this->minPLR ? partLoadRatio : 0.0);
+    this->loadSideEnergy = this->loadSideHeatTransfer * reportingInterval;
+
+    // calculate load side outlet conditions
+    Real64 const loadMCp = this->loadSideMassFlowRate * CpLoad;
+    this->loadSideOutletTemp = this->calcLoadOutletTemp(this->loadSideInletTemp, this->loadSideHeatTransfer / loadMCp);
+
+    // calculate power usage from EIR curves
+    Real64 eirModifierFuncTemp = Curve::CurveValue(state,
+                                                   this->powerRatioFuncTempCurveIndex,
+                                                   waterTempforCurve,
+                                                   oaTempforCurve); // CurveManager::CurveValue(state, this->powerRatioFuncTempCurveIndex,
+                                                                    // this->loadSideOutletTemp, this->sourceSideInletTemp);
+
+    if (eirModifierFuncTemp < 0.0) {
+        if (this->eirModFTErrorIndex == 0) {
+            ShowSevereMessage(state, format("{} \"{}\":", DataPlant::PlantEquipTypeNames[static_cast<int>(this->EIRHPType)], this->name));
+            ShowContinueError(state, format(" EIR Modifier curve (function of Temperatures) output is negative ({:.3T}).", eirModifierFuncTemp));
+            ShowContinueError(state,
+                              format(" Negative value occurs using a water temperature of {:.2T}C and an outdoor air temperature of {:.2T}C.",
+                                     waterTempforCurve,
+                                     oaTempforCurve));
+            ShowContinueErrorTimeStamp(state, " Resetting curve output to zero and continuing simulation.");
+        }
+        ShowRecurringWarningErrorAtEnd(state,
+                                       format("{} \"{}\": EIR Modifier curve (function of Temperatures) output is negative warning continues...",
+                                              DataPlant::PlantEquipTypeNames[static_cast<int>(this->EIRHPType)],
+                                              this->name),
+                                       this->eirModFTErrorIndex,
+                                       eirModifierFuncTemp,
+                                       eirModifierFuncTemp);
+        eirModifierFuncTemp = 0.0;
+    }
+
+    Real64 miniPLR_mod = 0.25; // 2022-05-17: maybe should use input minPLR; however, this is to duplicate the ems verson
```

Comment:
For sure this should use the input minPLR. There are other hard-wired values in this function that should be removed before this goes in.

Reply:
Addressed in earlier commits.